File: src/2025-02-24-deepclaude-rust/java/exercises/practice/resistor-color-trio/src/main/java/ResistorColorTrio.java
class ResistorColorTrio {
    private static final String[] COLOR_CODES = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };

    private int getValue(String color) {
        for (int i = 0; i < COLOR_CODES.length; i++) {
            if (COLOR_CODES[i].equals(color)) {
                return i;
            }
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }

    String label(String[] colors) {
        int first = getValue(colors[0]);
        int second = getValue(colors[1]);
        int exponent = getValue(colors[2]);

        long mainValue = (long) first * 10 + second;
        long multiplier = (long) Math.pow(10, exponent);

        long total = mainValue * multiplier;

        String suffix = "ohms";
        long divisor = 1;

        if (total >= 1_000_000_000L && total % 1_000_000_000L == 0) {
            divisor = 1_000_000_000L;
            suffix = "giga" + suffix;
        } else if (total >= 1_000_000L && total % 1_000_000L == 0) {
            divisor = 1_000_000L;
            suffix = "mega" + suffix;
        } else if (total >= 1_000L && total % 1_000L == 0) {
            divisor = 1_000L;
            suffix = "kilo" + suffix;
        }

        long value = total / divisor;

        return String.format("%d %s", value, suffix);
    }
}

File: src/2025-02-24-deepclaude-rust/javascript/exercises/practice/resistor-color-trio/resistor-color-trio.js
//
// This is only a SKELETON file for the 'Resistor Color Trio' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class ResistorColorTrio {
  constructor(colors) {
    this.colors = colors;
  }

  get label() {
    const colorCodes = {
      black: 0,
      brown: 1,
      red: 2,
      orange: 3,
      yellow: 4,
      green: 5,
      blue: 6,
      violet: 7,
      grey: 8,
      white: 9,
    };

    const [first, second, third] = this.colors;
    const firstVal = colorCodes[first];
    const secondVal = colorCodes[second];
    const exponent = colorCodes[third];

    if (firstVal === undefined || secondVal === undefined || exponent === undefined) {
      throw new Error('invalid color');
    }

    const mainValue = firstVal * 10 + secondVal;
    const totalOhms = mainValue * (10 ** exponent);

    const divisors = [
      { divisor: 1e9, prefix: 'giga' },
      { divisor: 1e6, prefix: 'mega' },
      { divisor: 1e3, prefix: 'kilo' },
    ];

    for (const { divisor, prefix } of divisors) {
      if (totalOhms >= divisor && totalOhms % divisor === 0) {
        const value = totalOhms / divisor;
        return `Resistor value: ${value} ${prefix}ohms`;
      }
    }

    return `Resistor value: ${totalOhms} ohms`;
  }
}
